src/main/java/uk/gov/justice/digital/delius/service/CourtService.java

```diff
@@ -14,19 +17,46 @@
 
 @Service
 @AllArgsConstructor(onConstructor = @__(@Autowired))
+@Slf4j
 public class CourtService {
     private final CourtRepository courtRepository;
+    private final LookupSupplier lookupSupplier;
+    private final FeatureSwitches featureSwitches;
 
+    @Transactional
     public Court updateCourt(String code, UpdateCourtDto court) {
-        // TODO check feature switch and update court
-        return Court.builder().code(code).courtName(court.getCourtName()).build();
+        final var courtEntity = getMostLikelyCourt(code).orElseThrow(() -> new NotFoundException(String
+            .format("Court %s not found", code)));
+        if (isAllowedToUpdate(code)) {
+            courtEntity.setBuildingName(court.getBuildingName());
+            courtEntity.setCourtName(court.getCourtName());
+            courtEntity.setCountry(court.getCountry());
+            courtEntity.setCourtType(lookupSupplier.courtTypeSupplier(court.getCourtTypeCode()).orElseThrow());
+            courtEntity.setCounty(court.getCounty());
+            courtEntity.setFax(court.getFax());
+            courtEntity.setLocality(court.getLocality());
+            courtEntity.setPostcode(court.getPostcode());
+            courtEntity.setStreet(court.getStreet());
+            courtEntity.setTelephoneNumber(court.getTelephoneNumber());
+            courtEntity.setTown(court.getTown());
+            courtEntity.setSelectable(court.isActive() ? "Y" : "N");
+        } else {
+            log.warn(String.format("This Court Update feature for %s is currently switched off", code));
```

Comment:
odd to do a `format` inside a `log.warn`

Reply:
yeah, I ALWAYS forget